server/bin/pbench-index.py

```diff
@@ -36,6 +37,28 @@
 from pbench.server.utils import rename_tb_link, quarantine
 
 
+class CustomError1(Exception):
+    pass
+
+
+class CustomError2(Exception):
+    pass
+
+
+def handler1(signum, frame):
+    raise CustomError1
+
+
+def handler2(signum, frame):
+    raise CustomError2()
+
+
+# Establish signal handler for INT, SIGHUP, TERM.
+# signal.signal(signal.SIGHUP, handler)
+signal.signal(signal.SIGINT, handler1)
+signal.signal(signal.SIGTERM, handler2)
```

Comment:
What happens if a SIGINT or SIGTERM arrives outside of the `try/except` where the exception is caught? I don't know that anything bad would happen, but I'd like to make sure that you have thought about that particular situation.

Reply:
At the time of 'continue' i.e. SIGINT. It will just continue with the next iteration. 
At the time of 'break' i.e. SIGTERM, it will execute the else in l.482 and the finally in l.485.

Reply:
Well, the signal handlers get activated early, so if a signal is delivered to the process when we are *NOT* inside the for-loop where the exceptions are handled, the handler will be called and the exception will be generated, but there will be no exception handler to handle it, so presumably the program will terminate abnormally. You can simulate that that's what happens by putting a sleep() at some place before the loop is entered and then delivering a signal to the process. I *think* that that's OK, but I'm wondering if it would be preferable to activate the signal handlers just before the for-loop, rather than at the beginning of the program. Just thinking aloud here: it's not clear to me that one way is "better" than the other.

Reply:
yes, I like the idea to declare signal handlers just before the for-loop. Although I am also not sure which would be better, I will try to get deeper into it.

Reply:
@ndokos I tested it. It is working fine